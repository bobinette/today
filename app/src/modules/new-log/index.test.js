import React from 'react';
import { shallow } from 'enzyme';

import { NewLogInput } from '.';

test('render the input', () => {
  const props = {
    content: 'Test',
    actions: {
      createLog: jest.fn(),
      updateContent: jest.fn(),
    },
  };

  const tree = shallow(<NewLogInput {...props} />);
  expect(tree).toMatchSnapshot();
});
