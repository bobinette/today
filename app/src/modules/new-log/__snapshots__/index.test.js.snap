// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the input 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewLogInput
    actions={
      Object {
        "createLog": [MockFunction],
        "updateContent": [MockFunction],
      }
    }
    content="Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TextareaAutosize
          className="NewLogInput__Content"
          onChange={[Function]}
          onHeightChange={[Function]}
          placeholder="Tell me all about your day..."
          useCacheForDOMMeasurements={false}
          value="Test"
        />,
        <div
          className="flex flex-align-right"
        >
          <button
            className="btn btn-primary btn-sm"
            onClick={[Function]}
          >
            Create
          </button>
        </div>,
      ],
      "className": "NewLogInput",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "NewLogInput__Content",
          "onChange": [Function],
          "onHeightChange": [Function],
          "placeholder": "Tell me all about your day...",
          "useCacheForDOMMeasurements": false,
          "value": "Test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-primary btn-sm"
            onClick={[Function]}
          >
            Create
          </button>,
          "className": "flex flex-align-right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create",
            "className": "btn btn-primary btn-sm",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Create",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextareaAutosize
            className="NewLogInput__Content"
            onChange={[Function]}
            onHeightChange={[Function]}
            placeholder="Tell me all about your day..."
            useCacheForDOMMeasurements={false}
            value="Test"
          />,
          <div
            className="flex flex-align-right"
          >
            <button
              className="btn btn-primary btn-sm"
              onClick={[Function]}
            >
              Create
            </button>
          </div>,
        ],
        "className": "NewLogInput",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "NewLogInput__Content",
            "onChange": [Function],
            "onHeightChange": [Function],
            "placeholder": "Tell me all about your day...",
            "useCacheForDOMMeasurements": false,
            "value": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary btn-sm"
              onClick={[Function]}
            >
              Create
            </button>,
            "className": "flex flex-align-right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create",
              "className": "btn btn-primary btn-sm",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Create",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
